vpr/src/route/check_route.cpp

```diff
@@ -500,8 +500,8 @@ void recompute_occupancy_from_scratch() {
 
     /* First set the occupancy of everything to zero. */
 
-    for (size_t inode_idx = 0; inode_idx < device_ctx.rr_nodes.size(); inode_idx++)
-        route_ctx.rr_node_route_inf[inode_idx].set_occ(0);
+    for (const RRNodeId& id : device_ctx.rr_graph.nodes())
+        route_ctx.rr_node_route_inf[(size_t)id].set_occ(0);
```

Comment:
Can you also add a comment here to remind developers that ``rr_node_route_inf`` should adapt RRNodeId ?